src/main.rs

```diff
@@ -109,107 +98,38 @@ fn cargo_expand() -> io::Result<i32> {
     let outfile = outdir.as_ref().map(|dir| dir.path().join("expanded"));
 
     // Build cargo command
-    let mut cmd = Command::new(cargo_binary());
-    cmd.args(&wrap_args(args.clone(), outfile.as_ref()));
+    let cargo_args = wrap_args(args.clone(), outfile.as_ref());
+    let mut cmd = duct::cmd(cargo_binary(), cargo_args);
 
     // Pipe to a tmp file to separate out any println output from build scripts
     if let Some(outfile) = outfile {
         let mut filter_cargo = Vec::new();
         filter_cargo.extend(args.iter().map(OsString::as_os_str));
         filter_cargo.push(OsStr::new("--filter-cargo"));
 
-        let _wait = cmd.pipe_to(shell!("cat"), Some(&filter_cargo))?;
-        run(cmd)?;
-        drop(_wait);
-
-        cmd = Command::new("cat");
-        cmd.arg(outfile);
-    }
-
-    // Pipe to rustfmt
-    let _wait = match which_rustfmt {
-        Some(ref fmt) => {
-            let args: Vec<_> = env::args_os().collect();
-            let mut filter_rustfmt = Vec::new();
-            filter_rustfmt.extend(args.iter().map(OsString::as_os_str));
-            filter_rustfmt.push(OsStr::new("--filter-rustfmt"));
-
-            Some((
-                cmd.pipe_to(shell!(fmt), None)?,
-                cmd.pipe_to(shell!("cat"), Some(&filter_rustfmt))?,
-            ))
-        }
-        None => None,
-    };
-
-    // Pipe to pygmentize
-    let _wait = match which_pygmentize {
-        Some(pyg) => Some(cmd.pipe_to(shell!(pyg "-l" "rust" "-O" "encoding=utf8"), None)?),
-        None => None,
-    };
-
-    run(cmd)
-}
-
-fn run(mut cmd: Command) -> io::Result<i32> {
-    cmd.status().map(|status| status.code().unwrap_or(1))
-}
+        cmd = cmd
+            .stderr_to_stdout()
+            .pipe(duct::cmd(filter_cargo[0], filter_cargo[1..].iter()));
+        cmd.run()?;
 
-#[cfg(unix)]
-struct Wait(Vec<Child>);
+        let pyg_cmd = which_pygmentize.map(|pyg|
+            cmd!(pyg, "-l", "rust", "-O", "encoding=utf8"));
 
-#[cfg(unix)]
-impl Drop for Wait {
-    fn drop(&mut self) {
-        for child in &mut self.0 {
-            if let Err(err) = child.wait() {
-                let _ = writeln!(&mut io::stderr(), "{}", err);
+        // Pipe to rustfmt and/or pygmentize
+        if let Some(ref fmt) = which_rustfmt {
+            // TODO: This was previously filtering stderr, why?
```

Comment:
I believe this was to filter out warnings from rustfmt, like that it failed to meet the line length maximum.

Would it be easy to pygmentize / print the outfile without rustfmt if we find that rustfmt failed to parse it?

Reply:
Yea, that'd be nice.  I was wondering about just not using pipes at all.  By using only temp files, I think it would simplify things and make it easier to ignore rustfmt/pygmentize errors.  The downside would be no output until it's done, but that might not be too bad.  What do you think?

Reply:
Sounds reasonable, let's do it that way.